lib/puppet/functions/find_module.rb

```diff
@@ -0,0 +1,41 @@
+# Finds an existing file from a module and returns its path.
+#
+# The argument to this function should be a module name String
+# For example, the reference `mysql` will search for the
+# directory `<MODULES DIRECTORY>/mysql` and return the first
+# found on the modulepath.
+#
+# This function can also accept:
+#
+# * Multiple String arguments, which will return the path of the **first** module
+#  found, skipping non existing modules.
+# * An array of module names, which will return the path of the **first** module
+#  found from the given names in the array, skipping non existing modules.
+#
+# The function returns `undef` if none of the given modules were found
+#
+# @since 5.4.0
+#
+Puppet::Functions.create_function(:find_module, Puppet::Functions::InternalFunction) do
```

Comment:
I think the name chosen for this function `find_module` is less than ideal. It conflicts with the the `find_xxx` methods we have in the `Puppet::Pal`  API. I would suggest a name that clearly indicates that this is about finding the directory of a specific module.